content/blog/workflowsets-0-0-1/index.Rmd

```diff
@@ -0,0 +1,316 @@
+
+The dimension reduction methods that we'll try are a few different flavors of PCA, partial least squares (PLS), independent component analysis (ICA), and multi-dimensional scaling (MDS). With the exception of PLS, these are unsupervised procedures since they do not take into account the outcome data. 
+
+```{r recipes}
+base_recipe <- 
+  recipe(water ~ ., data = meat_train) %>% 
+  step_zv(all_predictors()) %>% 
+  step_YeoJohnson(all_predictors()) %>% 
+  step_normalize(all_predictors())
+
+pca_recipe <- 
+  base_recipe %>% 
+  step_pca(all_predictors(), num_comp = tune())
+
+pca_kernel_recipe <- 
+  base_recipe %>% 
+  step_kpca_rbf(all_predictors(), num_comp = tune(), sigma = tune())
+
+pls_recipe <- 
+  base_recipe %>% 
+  step_pls(all_predictors(), outcome = vars(water), num_comp = tune())
+
+ica_recipe <- 
+  base_recipe %>% 
+  step_ica(all_predictors(), num_comp = tune())
+
+mds_recipe <-
+  base_recipe %>%
+  step_isomap(
+    all_predictors(),
+    num_terms = tune(),
+    neighbors = tune()
+  )
+```
+
+Note that several methods have tuning parameters that need to be optimized. For our application, we'll feed the new features into a model and use the root mean squared error (RMSE) as the metric to determine good values of these parameters. 
+
+The input into a workflow set is a named list of preprocessors and parsnip model specifications. We'll could the [parsnip RStudio add-in](https://parsnip.tidymodels.org/reference/parsnip_addin.html) to write out the model code. Linear regression and multivariate adaptive regression spline (MARS) models will be combined with the different preprocessors. 
+
+```{r models}
+lm_spec <- linear_reg() %>% set_engine("lm")
+mars_spec <- mars() %>% set_engine("earth") %>% set_mode("regression")
+```
+
+Finally, to make the workflow set, named lists are created and given as inputs into [`workflow_set()`](https://workflowsets.tidymodels.org/reference/workflow_set.html). Note that we added the original data as-is using a "plain" R formula. 
+
+```{r lists}
+preprocessors <-
+  list(
+    plain = water ~ .,
+    pca = pca_recipe,
+    pca_rbf = pca_kernel_recipe,
+    pls = pls_recipe,
+    ica = ica_recipe,
+    mds = mds_recipe
+  )
+
+models <- list(lm = lm_spec, mars = mars_spec)
+
+meat_wflow_set <- workflow_set(preprocessors, models, cross = TRUE)
+meat_wflow_set
+```
+
+The 12 rows are the different combinations of preprocessors and models.  As will be shown below, we can evaluate them via resampling using the [`workflow_map()`](https://workflowsets.tidymodels.org/reference/workflow_map.html) function. 
+
+However, a few of these recipes have tuning parameters that are somewhat difficult to optimize. We might want to have specific tuning parameter ranges for the kernel PCA and MDS recipes. We can accomplish that in the usual way (via `parameters()` and `update()`) then update the workflow set with these options so that they are used later. 
```

Comment:
```suggestion
However, a few of these recipes have tuning parameters that are somewhat difficult to optimize. We might want to have specific tuning parameter ranges for the kernel PCA and MDS recipes. We can accomplish that [in the usual way (via `parameters()` and `update()`)](https://www.tmwr.org/tuning.html#tuning-params-tidymodels) then update the workflow set with these options so that they are used later. 
```